### PR TITLE
docs: update CLA info

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Update CLA info to the latest.

Fix https://github.com/eslint/eslint/issues/15367

#### Is there anything you'd like reviewers to focus on?
No

### CODE REVIEW

**Review by aladdin-add — APPROVED** [review]
LGTM~

**Review by akgupta0777 — APPROVED** [review]
Signed the CLA with the link in the PR. Ready to merge

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ To report a security vulnerability in ESLint, please use our [HackerOne program]
 
 ## Contributing Code
 
-Please sign our [Contributor License Agreement](https://cla.js.foundation/eslint/eslint) and read over the [Pull Request Guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests).
+In order to submit code or documentation to an ESLint project, please sign our [Contributor License Agreement](https://eslint.org/docs/developer-guide/contributing/#signing-the-cla) and read over the [Pull Request Guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests).
```
Comment (nzakas) [inline CONTRIBUTING.md:22]: I think we need a better place to link to. Right now the CLA links to a paragraph that says to sign the CLA, which in turn links to a GitHub repo where you can’t sign the CLA.

**Review by nzakas — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (akgupta0777) [reply CONTRIBUTING.md:22]: You can sign CLA there by making a test PR if you are not covered it will provide a link to sign the CLA process.

Comment (akgupta0777) [reply CONTRIBUTING.md:22]: But I agree that if there was a valid CLA signing link which doesn't require sending a PR, then thats a great improvement.

Comment (snitin315) [reply CONTRIBUTING.md:22]: [This is the direct link](https://contributor.easycla.lfx.linuxfoundation.org/#/cla/project/3d9a9c15-cb4a-478d-a185-d8d2881dd914/user/f5e71bf5-4dd3-4e03-b70f-7459c4deb7df)

```suggestion
In order to submit code or documentation to an ESLint project, please sign our [Contributor License Agreement](https://contributor.easycla.lfx.linuxfoundation.org/#/cla/project/3d9a9c15-cb4a-478d-a185-d8d2881dd914/user/f5e71bf5-4dd3-4e03-b70f-7459c4deb7df) and read over the [Pull Request Guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests).
```

Comment (nzakas) [reply CONTRIBUTING.md:22]: That one is tied to your name. :-/ Let me see if I can figure it out.

**Comment by nzakas** [comment]
Okay, it looks like EasyCLA doesn’t have a link to directly sign. The best we can do is point people to https://cla.openjsf.org/ which has a copy of the CLA text.

We probably want to change the docs to just say you’ll be asked to sign our CLA when you send your first pull request but you can preview the CLA at this URL.

**Review by aladdin-add — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ To report a security vulnerability in ESLint, please use our [HackerOne program]
 
 ## Contributing Code
 
-Please sign our [Contributor License Agreement](https://cla.js.foundation/eslint/eslint) and read over the [Pull Request Guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests).
+In order to submit code or documentation to an ESLint project, you’ll be asked to sign our CLA when you send your first pull request but you can preview the CLA at this [URL](https://cla.openjsf.org/). Also, please read over the [Pull Request Guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests).
```
Comment (nzakas) [inline CONTRIBUTING.md:22]:
```suggestion
In order to submit code or documentation to an ESLint project, you’ll be asked to sign our CLA when you send your first pull request.  (Read more about the Open JS Foundation CLA process at https://cla.openjsf.org/.) Also, please read over the [Pull Request Guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests).
```

**Review by nzakas — CHANGES_REQUESTED** [review]
Just a little clean up

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ To report a security vulnerability in ESLint, please use our [HackerOne program]
 
 ## Contributing Code
 
-Please sign our [Contributor License Agreement](https://cla.js.foundation/eslint/eslint) and read over the [Pull Request Guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests).
+In order to submit code or documentation to an ESLint project, you’ll be asked to sign our CLA when you send your first pull request. (Read more about the Open JS Foundation CLA process at https://cla.openjsf.org/.) Also, please read over the [Pull Request Guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests).
```
Comment (snitin315) [inline CONTRIBUTING.md:22]:
```suggestion
In order to submit code or documentation to an ESLint project, you’ll be asked to sign our CLA when you send your first pull request. (Read more about the Open JS Foundation CLA process at <https://cla.openjsf.org/>.) Also, please read over the [Pull Request Guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests).
```

**Review by nzakas — APPROVED** [review]
LGTM

**Pull request merged**